app/services/chess_piece.rb

```diff
@@ -1,10 +1,11 @@
 class ChessPiece
   attr_accessor :position
   attr_reader :color
+
   BOARD_START = 0
   BOARD_END = 7
 
-  def initialize(color, position)
+  def initialize(color, position, html_icon=nil)
```

Comment:
Is html_icon used anywhere? Looks like it's a property of the subclass which is correct, but I don't see a need for it in base class.